File: 卡码/代码随想录/1/06-二叉树/Python/02-二叉树属性/LeetCode_0110.py
#110. 平衡二叉树
from typing import *
from ..TreeNode import *

class Solution:
  def getHeight(self,root:Optional[TreeNode])->int:
    if not root:
      return 0
    leftHeight:int=self.getHeight(root.left)
    rightHeight:int=self.getHeight(root.right)
    if leftHeight==-1 or rightHeight==-1:
      return -1
    return -1 if abs(leftHeight-rightHeight)>1 else 1+max(leftHeight,rightHeight)

  def isBalanced(self, root: Optional[TreeNode]) -> bool:
    return True if self.getHeight(root)!=-1 else False

File: 卡码/代码随想录/1/06-二叉树/Java/binaryTree/src/main/java/com/kurumi/binarytree/_02_二叉树属性/LeetCode_0110.java
//110. 平衡二叉树
package com.kurumi.binarytree._02_二叉树属性;

import com.kurumi.binarytree.TreeNode;

public class LeetCode_0110 {

}
class Solution {
  public int getHeight(TreeNode root){
    if(root==null) return 0;
    int leftHeight=getHeight(root.left);
    int rightHeight=getHeight(root.right);
    if(leftHeight==-1||rightHeight==-1) return -1;
    return Math.abs(leftHeight-rightHeight)>1 ? -1 : 1+Integer.max(leftHeight,rightHeight);
  }
  public boolean isBalanced(TreeNode root) {
    return getHeight(root)==-1 ? false : true;
  }
}
